www/pages/plan-trip/plan-trip-controller.js

```diff
@@ -156,121 +156,139 @@ angular.module('pvta.controllers').controller('PlanTripController', function ($s
-      if (!place.geometry) {
-        console.error('Place has no geometry.');
-        return;
-      }
-      if ($scope.bounds.contains(place.geometry.location)) {
-        expandViewportToFitPlace($scope.map, place);
-        $scope.params.destination.id = place.place_id;
-        $scope.params.destination.name = place.name;
+      mapLocation(destinationAutocomplete.getPlace(), function (place) {
+        $scope.request.destination.id = place.place_id;
+        $scope.request.destination.name = place.name;
+        //Name the trip: ORIGIN to DESTINATION if not destinationOnly, otherwise just DESTINATION
+        if ($scope.request.destinationOnly || !$scope.request.origin.name) {
+          $scope.request.name = place.name;
+        } else {
+          $scope.request.name = $scope.request.origin.name + ' to ' + place.name;
+        }
         $scope.$apply();
-      }
-      else {
-        $scope.params.destination.id = null;
-        destinationInput.value = $scope.params.destination.name;
-        invalidLocationPopup();
-      }
+      }, function (err) {
+        $scope.request.destination.id = null;
+        destinationInput.value = $scope.request.destination.name;
+      });
     });
   }
 
-
-  function expandViewportToFitPlace (map, place) {
-    if (place.geometry.viewpoint) {
-      map.fitBounds(place.geometry.viewpoint);
+  function mapLocation (place, success, error) {
+    if (!place.geometry || (place.geometry && !bounds.contains(place.geometry.location))) {
+      invalidLocationPopup();
+      error();
     } else {
-      map.setCenter(place.geometry.location);
-      map.setZoom(17);
+      //Fit the location on the map
+      if (place.geometry.viewpoint) {
+        $scope.map.fitBounds(place.geometry.viewpoint);
+      } else {
+        $scope.map.setCenter(place.geometry.location);
+        $scope.map.setZoom(17);
+      }
+      //success callback
+      success(place);
     }
   }
 
   /*
+   *
    * Uses all the trip params and
    * requests a trip from Google.
    */
   $scope.getRoute = function () {
     // We need an origin and destination
-    if (!$scope.params.origin.id || !$scope.params.destination.id) {
+    if (!$scope.request.origin.id || !$scope.request.destination.id) {
       return;
     }
-    $scope.route = {
-      directions: [],
-      arrivalTime: null,
-      departureTime: null,
-      origin: null,
-      destination: null
-    };
+
     // Google won't return trips for times past.
     // Instead of throwing an error, assume the user wants
     // directions for right now.
-    if ($scope.params.time.datetime < Date.now()) {
-      $scope.updateASAP(true);
+    if (!$scope.request.time.option.isASAP && $scope.request.time.datetime < Date.now()) {
+      $scope.request.time.option = $scope.timeOptions[0];
+      $ionicPopup.alert({
+        title: 'Requested Time In Past',
+        template: 'The trip you have requested is in the past. Route will find buses leaving now.'
```

Comment:
Dig it. Do we need to include seconds on the time though?  It wouldn't help me as a passenger haha